### PR TITLE
[BI-24] Create UI register page

### PR DESCRIPTION
**Pull request by LAlex14**
image

am creat ui ul pentru pagina de inregistrare, butonul submit este activ doar cand toate inputurile sunt valide, in momentul in care dam submit, se afișeaza în consolă obiectul cu datele introduse

### CODE REVIEW

File: src/views/InputComp.vue [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      <form @submit.prevent="handleSubmit">
+        <base-input
+          v-model="email"
+          type="email"
+          :name="$t('Email')"
+          rules="email|required"
+          :placeholder="$t('emailPH')"
+          @input="onInput"
+        >
+          {{ $t('emailLabel') }}
```
Comment (Ceciliaa) [inline src/views/InputComp.vue:16]: You can send the label value as prop 
`:label="$t('emailLabel')"`

File: src/views/InputComp.vue [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          {{ $t('Submit') }}
+        </button>
+      </form>
+    </ValidationObserver>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'inputTest',
```
Comment (Ceciliaa) [inline src/views/InputComp.vue:32]: The name property should  have the same value as the name of the component.  In this case it should be:
`InputComp`.
Another thing, the component name should be suggestive, we should be able to have an idea about the component content based on its name. 
InputComp doesn't tell me anything

File: src/views/Styleguide.vue [hunk 3]
```diff
@@ -1,10 +1,17 @@
 <template>
-  <div>{{ $t('styleguide') }}!</div>
-  <p class="text-red-300 font-bold">styleguide</p>
+  <div>
+    <div>{{ $t('styleguide') }}!</div>
+    <p class="font-bold text-red-300">styleguide</p>
```
Comment (Ceciliaa) [inline src/views/Styleguide.vue:4]: Missing Translation

File: src/App.vue [hunk 4]
```diff
@@ -1,13 +1,10 @@
 <template>
   <div id="app">
-    <div id="nav">
+    <div id="nav" class="flex justify-around">
       <router-link to="/">Home</router-link>
       <router-link to="/styleguide">Styleguide</router-link>
+      <router-link to="/register">Register</router-link>
```
Comment (Ceciliaa) [inline src/App.vue:6]: Missing translations

File: src/locales/en.json [hunk 5]
```diff
@@ -1,4 +1,17 @@
...
+  "Password": "Password",
+  "Password confirmation": "Password confirmation",
+  "password": "password",
+  "password confirmation": "password confirmation",
+  "BaseInput example": "BaseInput component (email)",
+  "emailPH": "john_kennedy@binarcode.com",
+  "emailLabel": "Email: ",
+  "Submit": "Submit",
+  "forgotPassText": "Forgot your password?",
+  "Sign Up": "Sign up"
```
Comment (Ceciliaa) [inline src/locales/en.json:16]: Let's change the keys:
`{
  "general.styleguide.tile": "Styleguide",
  "general.styleguide.section.baseInput": "BaseInput component (email)",
  "general.homePage.title": "Home page",
  
  
  "general.submit": "Submit",
  
  "register.label.firstName": "First Name",
  "register.label.lastName": "Last Name",
  "register.name.email": "Email",
  "register.label.email": "Email:",
  "register.label.password": "Password",
  "register.label.passwordConfirmation": "Password confirmation",
  "register.button.label.signUp": "Sign up",
  "register.placeholder.email": "john_kennedy@binarcode.com",
}`

**Review by Ceciliaa — APPROVED** [review]

File: src/views/Register.vue [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        {{ $t('Sign Up') }}
+      </h2>
+    </div>
+    <div class="mt-8 sm:mx-auto sm:w-full sm:max-w-md">
+      <div class="px-4 py-8 bg-white shadow sm:rounded-lg sm:px-10">
+        <ValidationObserver v-slot="{ valid }">
+          <form @submit.prevent="handleSubmit">
+            <base-input
+              v-model="model.first_name"
+              type="text"
```
Comment (PascuMihai) [inline src/views/Register.vue:21]: you can make the type text default so you can remove from here, just a tip

File: src/views/Register.vue [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+              :name="$t('register.name.firstName')"
+              :label="$t('register.name.firstName')"
+              rules="required"
+              placeholder="John"
+            />
+
+            <base-input
+              v-model="model.last_name"
+              type="text"
```
Comment (PascuMihai) [inline src/views/Register.vue:30]: same here

File: src/views/Register.vue [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+      <div class="px-4 py-8 bg-white shadow sm:rounded-lg sm:px-10">
+        <ValidationObserver v-slot="{ valid }">
+          <form @submit.prevent="handleSubmit">
+            <base-input
+              v-model="model.first_name"
+              type="text"
+              :name="$t('register.name.firstName')"
+              :label="$t('register.name.firstName')"
+              rules="required"
+              placeholder="John"
```
Comment (PascuMihai) [inline src/views/Register.vue:25]: placeholder First name and add translation

File: src/views/Register.vue [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+            />
+
+            <base-input
+              v-model="model.last_name"
+              type="text"
+              :name="$t('register.name.lastName')"
+              :label="$t('register.name.lastName')"
+              rules="required"
+              placeholder="Kennedy"
```
Comment (PascuMihai) [inline src/views/Register.vue:34]: placeholder Last name and translation

**Review by PascuMihai — APPROVED** [review]

**Pull request merged**